pyhf/prob/__init__.py

```diff
@@ -0,0 +1,46 @@
+from .. import get_backend
+import numpy as np
+
+class Simultaneous(object):
+    def __init__(self, factors, projections):
+        self.factors = factors
+        self.projections = projections
+        self.batch_shape = (sum([x.batch_shape[0] for x in self.factors]),)
+    
+    def sample(self, sample_shape = None):
+        tensorlib, _ = get_backend()
+        pieces = []
+        for fac in self.factors:
+            pieces.append(fac.sample(sample_shape if sample_shape is not None else (1,)))
+        result = tensorlib.concatenate(pieces,axis = -1) 
+        return result if sample_shape is not None else result[0]
+    
+    def log_prob(self, value):
```

Comment:
I guess the logpdf is calculated here now instead? There's a lot of changes/additions that I don't understand.